web/src/main/java/no/nav/modiapersonoversikt/consumer/norg/NorgApi.kt

```diff
@@ -40,40 +51,78 @@ interface NorgApi {
         diskresjonskode: String?
     ): List<Enhet>
 
-    fun hentKontaktinfo(enhet: String): EnhetKontaktinformasjon
+    fun hentKontaktinfo(enhet: EnhetId): EnhetKontaktinformasjon
 }
 
-class NorgApiImpl(url: String, httpClient: OkHttpClient) : NorgApi {
+class NorgApiImpl(
+    url: String,
+    httpClient: OkHttpClient,
+    scheduler: Timer = Timer(),
+    val clock: Clock = Clock.systemDefaultZone()
+) : NorgApi {
     private val log: Logger = LoggerFactory.getLogger(NorgApi::class.java)
+    private val cacheRetention = Duration.ofHours(1)
+    private var cache: Map<EnhetId, EnhetKontaktinformasjon> = emptyMap()
+    private var lastUpdateOfCache: LocalDateTime? = null
+    private val navkontorCache = createNorgCache<String, Enhet>()
+    private val gtCache = createNorgCache<String, List<EnhetGeografiskTilknyttning>>()
+
+    private val retry = Retry(
+        Retry.Config(
+            initDelay = 30 * 1000,
+            growthFactor = 2.0,
+            delayLimit = 60 * 60 * 1000,
+            scheduler = scheduler
+        )
+    )
+
     private val arbeidsfordelingApi = ArbeidsfordelingApi(url, httpClient)
     private val enhetApi = EnhetApi(url, httpClient)
-    private val enhetKontaktinfoApi = KontaktinformasjonApi(url, httpClient)
+    private val enhetKontaktInfoApi = EnhetskontaktinfoApi(url, httpClient)
+
+    init {
+        hentEnheterOgKontaktinformasjon()
+        scheduler.scheduleAtFixedRate(
+            delay = cacheRetention.toMillis(),
+            period = cacheRetention.toMillis(),
+            action = { hentEnheterOgKontaktinformasjon() }
+        )
+    }
```

Comment:
Forstår jeg det rett?: Vi henter all kontaktinformasjon for enhet hver time og populerer cachen, og deretter bare bruker cachen den neste timen?

Reply:
Jupp, det stemmer.